src/aura/BGE_BatchGiftEntryTab/BGE_BatchGiftEntryTabHelper.js

```diff
@@ -87,16 +100,17 @@
 
     /**
      * @description: checks that user has all necessary permissions and then launches modal or displays error
+     * @param: sourceBatchId Id of Batch record to be copied - optional
      */
-    checkFieldPermissions: function(component) {
+    checkFieldPermissions: function(component, sourceBatchId) {
         var action = component.get('c.checkFieldPermissions');
         action.setCallback(this, function (response) {
             var state = response.getState();
             if (state === 'SUCCESS') {
                 if (!component.get('v.modalOpen')) {
```

Comment:
Followup on above comment about passing in callback, this if block would become:
   onSuccess()

And we'd want to move the if(modalOpen) bit into the openNewBatchWizard function I think

Reply:
In my double click testing, the ifModalOpen bit had to go here or it wouldn't work (multiple modals could still be opened on a quick double click). It might be different if it's structured as a callback though?